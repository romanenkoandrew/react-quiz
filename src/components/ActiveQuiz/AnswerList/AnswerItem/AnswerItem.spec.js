import React from "react";
import { shallow } from "enzyme";
import AnswerItem from "./AnswerItem";

describe("AnswerItem component", () => {
  it("should render AnswerItem with props.state", () => {
    const component = shallow(
      <AnswerItem state={true} answer={{ text: "test", id: "test" }} />
    );
    expect(component).toMatchSnapshot();
  });
  it("should render AnswerItem without props.state", () => {
    const component = shallow(
      <AnswerItem state={false} answer={{ text: "test", id: "test" }} />
    );
    expect(component).toMatchSnapshot();
  });
  it("should clicked AnswerItem", () => {
    const mockCallBack = jest.fn();
    const component = shallow(
      <AnswerItem
        onAnswerClick={mockCallBack}
        answer={{ text: "test", id: "test" }}
      />
    );
    component.simulate("click");
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });
});
