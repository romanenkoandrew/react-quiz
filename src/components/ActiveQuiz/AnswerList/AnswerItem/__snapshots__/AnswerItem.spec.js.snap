// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnswerItem component should render AnswerItem with props.state 1`] = `
<li
  className="AnswerItem true"
  onClick={[Function]}
>
  test
</li>
`;

exports[`AnswerItem component should render AnswerItem without props.state 1`] = `
<li
  className="AnswerItem"
  onClick={[Function]}
>
  test
</li>
`;
